### PR TITLE
added Backnedless.Xamarin project

### PR DESCRIPTION
**Pull request by holailusoria**

### CODE REVIEW

File: Backendless/Messaging/DeviceRegistrationDto.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
-using System.Collections.Generic;
 using Weborb.Service;
+using Plugin.DeviceInfo;
```
Comment (markpiller) [inline Backendless/Messaging/DeviceRegistrationDto.cs:3]: Should there be a conditional compiler statement (#IF MOBILE for example) ?

File: Backendless/Push/Registrar.cs [hunk 2]
```diff
@@ -19,9 +20,8 @@ internal static String RegisterDevice( String token, List<String> channels, Date
       deviceRegistration.Channels = channels;
       deviceRegistration.Expiration = expiration;
       deviceRegistration.DeviceToken = token;
-      deviceRegistration.DeviceId = Backendless.Messaging.DeviceID;
       deviceRegistration.Channels = channels;
-      deviceRegistration.Os = DeviceCheck.GetDeviceOS();
+      deviceRegistration.Os = DeviceCheck.GetDeviceOS();
```
Comment (markpiller) [inline Backendless/Push/Registrar.cs:24]: all these things with mobile specific functionality must have conditional compiler statements, so this code doesn't break other builds

File: Backendless/Service/MessagingService.cs [hunk 3]
```diff
@@ -72,13 +77,12 @@ public MessagingService()
 #endif
     }
 
-#if UNITY_ANDROID || UNITY_IPHONE || MOBILE
+#if (UNITY_ANDROID || UNITY_IPHONE || MOBILE) && NETSTANDARD2_1
```
Comment (markpiller) [inline Backendless/Service/MessagingService.cs:80]: Why NETSTANDARD2_1? Plus it looks like .net standard will be going away: https://prnt.sc/yysixh
read more about it here: https://docs.microsoft.com/en-us/dotnet/standard/net-standard

**Review by markpiller — CHANGES_REQUESTED** [review]

File: Backendless/Push/Registrar.cs (hunk 2 above)
Comment (holailusoria) [reply Backendless/Push/Registrar.cs:24]: All other projects doesn't have reference to this class

File: Backendless/Messaging/DeviceRegistrationDto.cs (hunk 1 above)
Comment (holailusoria) [reply Backendless/Messaging/DeviceRegistrationDto.cs:3]: All other projects doesn't have reference to this class

File: Backendless/Service/MessagingService.cs (hunk 3 above)
Comment (holailusoria) [reply Backendless/Service/MessagingService.cs]: Fixed

File: Backendless/WebORB/Reader/ArrayType.cs [hunk 4]
```diff
@@ -196,13 +196,13 @@ public object adapt( Type type, ReferenceCache refCache )
 
         object newDictionary = Activator.CreateInstance( constructedType );
         refCache.AddObject( this, type, newDictionary );
-        Type keyValuePairType = typeof( KeyValuePair );
+        Type keyValuePairType = typeof( Weborb.Util.KeyValuePair );
```
Comment (markpiller) [inline Backendless/WebORB/Reader/ArrayType.cs:199]: Why is this change needed?

File: Projects/Backendless.Xamarin/LocalStorage.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        }
+      }
+      private set => _deviceId = String.IsNullOrEmpty( _deviceId ) ? value : _deviceId;
+    }
+
+    public void SaveData()
+    {
+      try
+      {
+        String fileName = Path.Combine( path, "BackendlessDeviceId.txt" );
```
Comment (markpiller) [inline Projects/Backendless.Xamarin/LocalStorage.cs:44]: Why does a class with a generic name "LocalStorage" is hardcoded to worj with "BackendlessDeviceId.txt ?
Why do we need this class?

**Review by markpiller — CHANGES_REQUESTED** [review]

File: Backendless/WebORB/Reader/ArrayType.cs (hunk 4 above)
Comment (holailusoria) [reply Backendless/WebORB/Reader/ArrayType.cs:199]: Conflict between System.Collections.Generic.KeyValuePair and Weborb.Util.KeyValuePair

File: Projects/Backendless.Xamarin/LocalStorage.cs (hunk 5 above)
Comment (holailusoria) [reply Projects/Backendless.Xamarin/LocalStorage.cs:44]: These are unnecessary files. Have been removed.

File: Backendless/WebORB/Reader/ArrayType.cs (hunk 4 above)
Comment (markpiller) [reply Backendless/WebORB/Reader/ArrayType.cs:199]: why is there a conflict now? did you change "using" statements in "ArrayType"?

**Review by markpiller — CHANGES_REQUESTED** [review]

File: Backendless/WebORB/Reader/ArrayType.cs (hunk 4 above)
Comment (holailusoria) [reply Backendless/WebORB/Reader/ArrayType.cs]: In Netstandard 2.1, the "KeyValuePair" class could be used without generics.
Because we're back to 2.0 - this was fixed in the last commit.
